src/selectBetweenOffsets.ts

```diff
@@ -0,0 +1,81 @@
+import * as vscode from "vscode";
+import { ExtensionHostMessageHandler, MessageType } from "../shared/protocol";
+import { ISelectionState } from "./hexDocument";
+import { HexEditorRegistry } from "./hexEditorRegistry";
+
+const addressRe = /^0x[a-f0-9]+$/i;
+const decimalRe = /^[0-9]+$/i;
+
+export const showSelectBetweenOffsets = (messaging: ExtensionHostMessageHandler, registry: HexEditorRegistry): void => {
+    const startingInput = vscode.window.createInputBox();
+    const endingInput = vscode.window.createInputBox();
+    startingInput.placeholder = "Enter offset to select from";
+    endingInput.placeholder = "Enter offset to select until";
+
+    messaging.sendEvent({ type: MessageType.StashDisplayedOffset });
+
+    let inputOffset: number | undefined;
+    let fromOffset: number | undefined;
+    let toOffset: number | undefined;
+    let accepted = false;
+
+    // acquire selection state from active HexDocument
+    const selectionState: ISelectionState | undefined = registry.activeDocument?.selectionState;
+
+    // if there is a selection, use the focused offset as the starting offset
+    if (selectionState !== undefined && selectionState.selected > 0 && selectionState.focused !== undefined) {
+        inputOffset = selectionState.focused;
+        // converting to hex to increase readability
+        startingInput.value = `0x${inputOffset.toString(16)}`;
+    }
+
+    startingInput.onDidChangeValue(changeValueHelper);
+    endingInput.onDidChangeValue(changeValueHelper);
+
+    startingInput.onDidHide(hideHelper);
+    endingInput.onDidHide(hideHelper);
+
+    startingInput.onDidAccept(() => {
+        if (inputOffset !== undefined) {
+            fromOffset = inputOffset;
+            endingInput.show();
+        }
+    });
+    endingInput.onDidAccept(() => {
+        if (inputOffset !== undefined) {
+            toOffset = inputOffset;
+        }
+
+        if (fromOffset !== undefined && toOffset !== undefined) {
+            accepted = true;
+            messaging.sendEvent({ type: MessageType.SetFocusedByteRange, startingOffset: fromOffset, endingOffset: toOffset });
+        }
+    });
+
+    startingInput.show();
+
+    function hideHelper() {
```

Comment:
you'll want to make sure to `.dispose()` each input after their used.

Reworking this function to use some inner `function getOffset(options: { ... }): Promise<number>` would probably be easier to deal with

Reply:
I thought `onDidHide` already disposes the `InputBox`. In `goToOffset.ts` file, there isn't a `.dispose()` for the input. And if I call `.dispose()` in `hideHelper` function, `hideHelper` gets called twice. Should I still call `.dispose()`?

Reply:
Yes, we should always call `dispose`. There are additional resources that get collected.

Reply:
I did what you suggested and I think it turned out great. Let me know if there are any problems.